src/main/java/com/fundynamic/d2tm/game/gui/BuyStuffGuiElement.java

```diff
@@ -0,0 +1,168 @@
+
+        // 3 icons in a row. We assume the dimensions of the first one
+        widthOfIcon = images.get(0).getWidth();
+        heightOfIcon = images.get(0).getHeight();
+
+        // the height of this window is:
+        // 26 pixels for the 'top' (ie, the title of the thing, Const Yard, etc)
+        // then we add the height + 1 pixel room (in between the icons)
+        int rows = 1 + (int) Math.floor(images.size() / 4);
+
+        int heightOfWindow = 26 + (rows * heightOfIcon) + rows + 1;
+        int widthOfWindow =  1 + (4 * widthOfIcon) + 3 + 1; // 1 pixel left border + 4 icons + 3 times in between a pixel + 1 pixel for border at the right
+
+        rectangle = new Rectangle(x, y, x + widthOfWindow, y + heightOfWindow);
+
+        System.out.println("The gui window dimensions are: " + rectangle);
+
+
+        // Window content
+        int imageX = getX() + 1; // 1 pixel is the border
+        int imageY = getY() + 26; // start beneath the 'title bar'
+
+        int horizontalIconCounter = 0;
+        for (int i = 0; i < images.size(); i++) {
+            Image image = images.get(i);
+
+            BuyIcon buyIcon = new BuyIcon(imageX, imageY, image);
+            buyIconList.add(buyIcon);
+            horizontalIconCounter++;
+
+            imageX += widthOfIcon;
+
+            if (horizontalIconCounter > 3) {
+                horizontalIconCounter = 0;
+
+                // row is full
+                imageY += heightOfIcon;
+
+                // as long as expect a full new row to be rendered, add another extra pixel so we keep
+                // a pixel space between icons
+                if ((i + 1) < images.size()) { // 1 icon is enough for a row
+                    imageY++;
+                }
+                imageX = getX() + 1; // reset X again
+            } else {
+                imageX++; // and one pixel in between
+            }
+        }
+    }
+
+    public int getX() {
+        return rectangle.getTopLeftXAsInt();
+    }
+
+    public int getY() {
+        return rectangle.getTopLeftYAsInt();
+    }
+
+    public int getMostRightXCoordinate() {
+        return rectangle.getBottomRight().getXAsInt();
+    }
+
+    public int getMostDownYCoordinate() {
+        return rectangle.getBottomRight().getYAsInt();
+    }
+
+    private int getHeightAsInt() {
+        return rectangle.getHeightAsInt();
+    }
+
+    private int getWidthAsInt() {
+        return rectangle.getWidthAsInt();
+    }
+
+    public boolean isVectorWithin(Vector2D vec) {
+        return rectangle.isVectorWithin(vec);
+    }
+
+    @Override
+    public void render(Graphics graphics, int x, int y) {
+        graphics.setColor(Color.black);
+        graphics.fillRect(getX(), getY(), getWidthAsInt(), getHeightAsInt());
+
+        graphics.setColor(Color.darkGray);
+        graphics.drawRect(getX(), getY(), getWidthAsInt(), getHeightAsInt());
+
+        // title
+        graphics.setAntiAlias(false);
```

Comment:
Does it make sense to disable anti-aliasing here? It's not being reset to `true` afterwards. Does this affect rendering for other elements? If not, then I'd expect this to be a global setting somewhere in the main render loop.